src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -112,9 +132,152 @@ def export_helm_chart(registry_path, chart_export_path, kube_config, kube_contex
     response_helm_chart_export = subprocess.Popen(cmd_helm_chart_export, stdout=PIPE, stderr=PIPE)
     _, error_helm_chart_export = response_helm_chart_export.communicate()
     if response_helm_chart_export.returncode != 0:
-        telemetry.set_exception(exception=error_helm_chart_export.decode("ascii"), fault_type=consts.Export_HelmChart_Fault_Type,
-                                summary='Unable to export helm chart from the registry')
-        raise CLIInternalError("Unable to export helm chart from the registry '{}': ".format(registry_path) + error_helm_chart_export.decode("ascii"))
+        if for_preonboarding_checks:
+            telemetry.set_exception(exception=error_helm_chart_export.decode("ascii"), fault_type=consts.Pre_Onboarding_Inspector_Export_HelmChart_Fault_Type,
+                                    summary='Unable to export pre-onboarding-inspector helm chart from the registry')
+            raise CLIInternalError("Unable to export pre-onboarding-inspector helm chart from the registry '{}': ".format(registry_path) + error_helm_chart_export.decode("ascii"))
+        else:
+            telemetry.set_exception(exception=error_helm_chart_export.decode("ascii"), fault_type=consts.Export_HelmChart_Fault_Type,
+                                    summary='Unable to export helm chart from the registry')
+            raise CLIInternalError("Unable to export helm chart from the registry '{}': ".format(registry_path) + error_helm_chart_export.decode("ascii"))
+
+
+def check_cluster_DNS(dns_check_log, for_preonboarding_checks=False, filepath_with_timestamp=None, storage_space_available=False):
+    if for_preonboarding_checks:
+        try:
```

Comment:
there is still a lot of repetition of code here. Only the uncommon part like storagespace etc should be extra based on the condition of whether it is for pre-onboarding check or diagnoser check